master/pkg/tasks/task.go

```diff
@@ -228,6 +228,7 @@ func TrialEnvVars(t TaskSpec, rendezvousPorts []string, tPortOffset int) map[str
 
 	envVars := defaultEnvVars()
 	envVars["DET_EXPERIMENT_ID"] = fmt.Sprintf("%d", exp.InitialWorkload.ExperimentID)
+	envVars["DET_TASK_ID"] = t.TaskID
```

Comment:
It doesn't seem like it is a _ton_ of work to generate a token and user it here instead. Task ID is exposed in:
* The `/tasks` endpoint.
* The names of k8s pods and in the labels of pods specs and config maps created by the master. 
* The fitting function in the logs it when it can't fit a task.
I think this is probably just as bad as not having auth.

Reply:
Logs and k8s are the ones you potentially don't need Determined auth to even see.

Reply:
We could basically duplicate all the user session logic and have `TaskSessions`:
```
// UserSession corresponds to a row in the "user_sessions" DB table.
type TaskSession struct {
	ID     SessionID `db:"id" json:"id"`
	TaskId string    `db:"task_id" json:"task_id"`
	Expiry time.Time `db:"expiry" json:"expiry"`
}
```

Reply:
Hmm, yeah, using the literal task ID may be a bit too exposed. Shiyuan, what do you think?

Reply:
Changed the PR to use task tokens that are assigned before starting task containers.